File: bot_discord.py
import discord
import requests
import asyncio

DISCORD_TOKEN = ""
CHATPDF_API_KEY = ""
SOURCE_ID = ""

def get_chatpdf_response(question):
    url = "https://api.chatpdf.com/v1/chats/message"
    headers = {
        "x-api-key": CHATPDF_API_KEY,
        "Content-Type": "application/json",
    }
    data = {
        "sourceId": SOURCE_ID,
        "messages": [
            {
                "role": "user",
                "content": question,
            }
        ]
    }

    response = requests.post(url, json=data, headers=headers)
    response.raise_for_status()
    return response.json()["content"]

intents = discord.Intents.default()
intents.message_content = True
client = discord.Client(intents=intents)

@client.event
async def on_ready():
    print(f'Bot conectado como {client.user}')

@client.event
async def on_message(message):
    if message.author == client.user:
        return

    if message.content.startswith('!pergunta'):
        def support_check(m):
            return m.author == message.author and m.channel == support_channel

        category_id = "858001239999250462"  # Substitua pelo ID da categoria "chamados - bot"

        await message.channel.send("Você será encaminhado para o chat suporte.")

        try:
            await asyncio.sleep(1)  # Pequeno atraso antes de criar o novo canal
            support_channel = await message.guild.create_text_channel("chat-suporte", category=message.guild.get_channel(int(category_id)))
            await support_channel.send("Qual sua dúvida? Você tem 5 minutos para fazer sua pergunta.")


            question_message = await client.wait_for('message', check=support_check, timeout=300)
            question = "Responda em português, de uma maneira simplificada como se fosse um funcionário de suporte: " + question_message.content.strip()

            # Adicionar a linha abaixo aqui
            chatpdf_response = get_chatpdf_response(question)

            while True:
                # Obter a resposta do ChatPDF
                chatpdf_response = get_chatpdf_response(question)

                # Enviar a resposta no mesmo canal "chat-suporte"
                await support_channel.send(chatpdf_response)

                # Esperar pela próxima mensagem do usuário no canal "chat-suporte"
                question_message = await client.wait_for('message', check=support_check, timeout=300)
                question = "Responda em português, de uma maneira simplificada como se fosse um funcionário de suporte, pode ser bem simplista rapido e direto, se der para não se exceder muito, à menos que seja um duvida que exija mais especifidade: " + question_message.content.strip()

                # Verificar se a mensagem começa com "!encerrar"
                if question_message.content.startswith('!encerrar'):
                    await message.channel.send("O chat-suporte foi encerrado a pedido do usuário.")
                    await asyncio.sleep(1)  # Pequeno atraso antes de excluir o canal
                    await support_channel.delete()
                    break

        except asyncio.TimeoutError:
            await message.channel.send("Tempo limite excedido. O chat-suporte foi encerrado.")
            await asyncio.sleep(1)  # Pequeno atraso antes de excluir o canal
            await support_channel.delete()

client.run(DISCORD_TOKEN)

File: testejs/bot_discord.js
const { Client, Intents } = require('discord.js');
const axios = require('axios');

const DISCORD_TOKEN = "MTEzMTU3MDcxOTk2OTUyMTY5NA.GbGg0E.iEg5iDhkYsuyGmADP4NXTwHv0b4S0l4GKqBE2Q";
const CHATPDF_API_KEY = "sec_MREH39JMvKwBuEg1zmKvkTkErqbvFYTa";
const SOURCE_ID = "share/m0ZVtQHsZ5AfAuLYjVaJF";

const intents = new Intents([
  Intents.FLAGS.GUILDS,
  Intents.FLAGS.GUILD_MESSAGES,
  Intents.FLAGS.DIRECT_MESSAGES,
]);

const client = new Client({ intents });

client.once('ready', () => {
  console.log('Bot conectado como ' + client.user.tag);
});

client.on('messageCreate', async (message) => {
  if (message.author.bot) return;

  if (message.content.startsWith('!pdf')) {
    await message.channel.send("Por favor, envie a pergunta que deseja fazer.");

    const filter = (m) => m.author.id === message.author.id;
    const questionMessage = await message.channel.awaitMessages({
      filter,
      max: 1,
      time: 60000,
      errors: ['time']
    });

    const question = questionMessage.first().content.trim();
    const response = await chatPdf(question, SOURCE_ID);

    await message.channel.send(response);
  }
});

async function chatPdf(question, sourceId) {
  const url = "https://api.chatpdf.com/v1/chats/message";
  const headers = {
    "x-api-key": CHATPDF_API_KEY,
    "Content-Type": "application/json",
  };
  const data = {
    "sourceId": sourceId,
    "messages": [
      {
        "role": "user",
        "content": question,
      }
    ]
  };

  try {
    const response = await axios.post(url, data, { headers });
    if (response.status === 200) {
      return response.data.content;
    } else {
      return "Desculpe, não consegui obter uma resposta.";
    }
  } catch (error) {
    console.error(error);
    return "Ocorreu um erro ao obter a resposta.";
  }
}

client.login(DISCORD_TOKEN);
